dpar/src/features/input_layers.rs

```diff
@@ -299,17 +302,48 @@ impl InputVectorizer {
                     address: vec![addr.dependent],
                     layer: addr::Layer::Token,
                 };
+                let addr_head_pos = addr::AddressedValue {
+                    address: vec![addr.head],
+                    layer: addr::Layer::Tag,
+                };
+                let addr_dependent_pos = addr::AddressedValue {
+                    address: vec![addr.dependent],
+                    layer: addr::Layer::Tag,
+                };
                 let head = addr_head.get(state);
                 let dependent = addr_dependent.get(state);
-                if let (Some(head), Some(dependent)) = (head, dependent) {
-                    let association = self.assoc_strength(&head, &dependent, &deprel);
+                let head_pos = addr_head_pos.get(state);
+                let dependent_pos = addr_dependent_pos.get(state);
+
+                if let (Some(head), Some(dependent), Some(head_pos), Some(dependent_pos)) =
+                    (head, dependent, head_pos, dependent_pos)
+                {
+                    let association =
+                        self.assoc_strength(&head, &dependent, &head_pos, &dependent_pos, &deprel);
                     non_lookup_slice[idx] = association;
                 }
             }
         }
     }
 
-    fn assoc_strength(&self, head: &str, dependent: &str, deprel: &str) -> f32 {
+    fn assoc_strength(
+        &self,
+        head: &str,
+        dependent: &str,
+        head_pos: &str,
+        dependent_pos: &str,
+        deprel: &str,
+    ) -> f32 {
+        let mut head = head.to_string();
+        let mut dependent = dependent.to_string();
+
+        if !self.no_lowercase_tags.contains(&head_pos.to_string()) {
```

Comment:
I think you had a question about this last week. With just two tags, a `Vec` is usually going to be faster than a set. So, I think it is justified over a hash or binary tree set.

Reply:
Yes, thanks, that's good to know 😊